### ISSUE TITLE
Unable to delete unvalidated line from accounting journal

### ISSUE THREAD

**Issue by dbeniamine** [Dolibarr/dolibarr#29690]
### Bug

I am not able to delete a line from a bookeeping journal, Dolibarr tells me it is already validated while it's not

### Dolibarr Version

18.0.5

### Environment PHP

7.4.33

### Environment Database

mariadbd  Ver 11.3.2-MariaDB-1:11.3.2+maria~ubu2204 for debian-linux-gnu on x86_64

### Steps to reproduce the behavior and expected behavior

Save a bookeeping line into a journal then try to delete it from the journal

The following path fixes the issues for me and seems more logical :+1: 

```patch
diff --git a/htdocs/accountancy/bookkeeping/list.php b/htdocs/accountancy/bookkeeping/list.php
index a8956a5e9a9..b6d468fe023 100644
--- a/htdocs/accountancy/bookkeeping/list.php
+++ b/htdocs/accountancy/bookkeeping/list.php
@@ -479,7 +479,7 @@ if (empty($reshook)) {
                                        setEventMessages($object->error, $object->errors, 'errors');
                                        $error++;
                                        break;
-                               } elseif (isset($object->date_validation) || $object->date_validation != '') {
+                               } elseif (isset($object->date_validation) && $object->date_validation != '') {
                                        setEventMessages($langs->trans("ValidatedRecordWhereFound"), null, 'errors');
                                        $error++;
                                        break;

```

**Comment by dbeniamine** [comment]
I think there is an error in my deployment, while trying to write an MR I realised that this is the upstream code, probably that an update failed on my part, I'm closing this issue

**Comment by dbeniamine** [comment]
Well it was fixed by 77afadcc1f8a6fea029ed75df2fbe3a64f2f637a which is in the branch 18.0 but included in the 18.0.5 release, so I guess it will be fixed in 18.0.6
